components/OutcropListPage/index.js

```diff
@@ -39,29 +44,31 @@ const OutcropListPage = ({ fieldtripId }) => {
     }
   }, []);
 
-  const [showPopup, setShowPopup] = useState(false);
-  const [newOutcropTitle, setNewOutcropTitle] = useState("");
-
   const handleBack = () => {
+    // Go back to previous page
     router.back();
   };
 
   const handleAddOutcrop = () => {
+    // Add new outcrop to current field trip
     if (newOutcropTitle.trim() !== "") {
+      // Check if new outcrop title is not empty
       try {
         const fieldTrips = JSON.parse(localStorage.getItem("fieldTrips"));
 
         const currentFieldTrip = fieldTrips.find(
+          // Find current field trip
           (fieldtrip) => fieldtrip.id === currentFieldTripId
         );
 
         const newOutcrop = {
+          // Create new outcrop
           id: currentFieldTrip.outcrops.length + 1,
           name: newOutcropTitle,
           details: [],
         };
 
-        currentFieldTrip.outcrops.push(newOutcrop);
+        currentFieldTrip.outcrops.push(newOutcrop); // Add new outcrop to current field trip
```

Comment:
you can use the newer [spread-syntax](https://github.com/neuefische/muc-web-23-3/blob/main/sessions/js-modern-syntax/js-modern-syntax.md#spread-syntax-) here.